x-pack/legacy/plugins/index_management/public/application/store/reducers/detail_panel.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { handleActions } from 'redux-actions';
+import {
+  UIM_DETAIL_PANEL_SUMMARY_TAB,
+  UIM_DETAIL_PANEL_SETTINGS_TAB,
+  UIM_DETAIL_PANEL_MAPPING_TAB,
+  UIM_DETAIL_PANEL_STATS_TAB,
+  UIM_DETAIL_PANEL_EDIT_SETTINGS_TAB,
+} from '../../../../common/constants';
+import {
+  TAB_SUMMARY,
+  TAB_SETTINGS,
+  TAB_MAPPING,
+  TAB_STATS,
+  TAB_EDIT_SETTINGS,
+} from '../../constants';
+import { openDetailPanel, closeDetailPanel } from '../actions/detail_panel';
+import { loadIndexDataSuccess } from '../actions/load_index_data';
+import {
+  updateIndexSettingsSuccess,
+  updateIndexSettingsError,
+} from '../actions/update_index_settings';
+import { deleteIndicesSuccess } from '../actions/delete_indices';
+
+const defaultState = {};
+
+export const getDetailPanelReducer = uiMetricService =>
```

Comment:
The diff on this file is to wrap the reducer inside a function and provide the uiMetricService. The rest of the file was left untouched.